parser/src/json/schema.rs

```diff
@@ -128,6 +131,40 @@ impl Schema {
         }
     }
 
+    pub fn const_compile(&self) -> Option<RegexNode> {
+        let str = match self {
+            Schema::Null => "null",
+            Schema::Boolean => return Some(RegexNode::Regex("true|false".to_string())),
+            Schema::Number {
+                minimum: Some(x),
+                maximum: Some(y),
+                ..
+            } if x == y => return Some(RegexNode::Literal(x.to_string())),
+            Schema::String {
+                const_string: Some(s),
+                ..
+            } => s,
```

Comment:
As long as you ensure that there are no other constraints (e.g. due to intersection with siblings) that might invalidate this const_str. Could maybe handle that in the intersection logic, but it should be sufficient to change this to

```rust
            Schema::String {
                min_length: 0,
                max_length: None,
                const_string: Some(s),
                pattern: None,
                format: None,
            } => s,
```

Reply:
Passes guidance tests (which include some weird cases where other constraints are combined with enums) with this change

Reply:
but of course this is totally mitigating the entire point of the const, as we're adding a regex when we add the const string... needs modification

Reply:
my idea was that `const_string` means it's just this `const_string`, but it's just a side information, it's also stored as `pattern`; this way the existing intersection logic still works

Reply:
but actually, we can just get rid of const_string completely, and *always* const_compile the ::String!

Reply:
it should be probably regex_compile not const_compile